lib/templatesimplifier.cpp

```diff
@@ -368,16 +388,36 @@ unsigned int TemplateSimplifier::templateParameters(const Token *tok)
     if (Token::Match(tok->previous(), "%var% <"))
         return 0;
     tok = tok->next();
+    if (tok->str() == ">")
+        return 0;
 
     unsigned int level = 0;
 
     while (tok) {
         // skip template template
         if (level == 0 && Token::simpleMatch(tok, "template <")) {
             const Token *closing = tok->next()->findClosingBracket();
-            if (closing)
+            if (closing) {
+                if (closing->str() == ">>") {
+                    if (level == 0)
```

Comment:
This condition is always true, isn't it? Because previous condition `if (level == 0 &&` on line 398.

Reply:
Yes. I'll fix it.

Reply:
Should cppcheck have found this?

Reply:
Yes it really should I am surprised it didn't. Thanks.

Reply:
I've created https://trac.cppcheck.net/ticket/9188 for this FN

Reply:
Thanks amai!

Reply:
Just a guess is that it assumes that the function call is changing variables behind our back.. if that is the case we can probably make cppcheck more clever about function calls.